py4j-python/release_process.txt

```diff
@@ -1,31 +1,40 @@
 0. Update all versions (version.py, properties, documentation, index.rst).
 1. Write changelog
-2. gradle clean
+2. gradle clean (at py4j/py4j-java)
 
 3. close all bugs on github
 4. tag the release on git
 
-5. gradle buildPython
-6. create signature using gpg (gpg -b --armor)
+5. gradle buildPython  (at py4j/py4j-java)
+6. create signature using gpg (gpg --export --armor YOUR_KEY, also refer to 'Preparing gpg key' in https://spark.apache.org/release-process.html)
 
-7. upload to PyPI using twine
+7. upload to PyPI using twine (at py4j/py4j-python). For example:
 
-8. mvn package javadoc:jar source:jar
-9. create signature for the three jar files and the pom (gpg -b --armor py4j-....jar)
-10. jar -cvf bundle.jar pom.xml ... (all files)
-11. create staging release in central (https://oss.sonatype.org Staging Upload, Upload Bundle)
-12. close release
-13. publish release
+    python setup.py sdist
+    python setup.py bdist_wheel --universal
+    twine upload dist/*
 
-14. gradle updateSite
-15. Extract and modify artifacts.xml and content.xml to change build path to:
+8. mvn clean deploy (at py4j/py4j-java)
+9. make sure you have 'settings.xml' under '~/.m2', see also https://central.sonatype.org/publish/publish-maven/#distribution-management-and-authentication
+10. if you are new users in Sonatype, you should replace the upload URL in pom.xml (at py4j/py4j-java). For example:
+
+    # In Mac
+    sed -i '' 's/oss.sonatype.org/s01.oss.sonatype.org/g' pom.xml
+    # or in Linux OS
+    sed 's/oss.sonatype.org/s01.oss.sonatype.org/g' pom.xml
+
+9. close release (https://oss.sonatype.org for old users, https://s01.oss.sonatype.org for new users)
+10. publish release (https://oss.sonatype.org for old users, https://s01.oss.sonatype.org for new users)
```

Comment:
@bartdag I have finished up to 10. .. Do you mind helping the Elapse plugin release? Basically, I couldn't follow "Extract and modify artifacts.xml and content.xml to change build path to" ....